packages/state/src/hooks/setAccountData.ts

```diff
@@ -53,20 +53,23 @@ export const useRestoreFromMnemonic = () => {
     derivationPathTemplate,
     label,
     curve,
+    mode = "mnemonic",
   }: {
     mnemonic: string;
     password: string;
     derivationPathTemplate: string;
     label: string;
     curve: Curves;
+    mode?: "mnemonic" | "new-mnemonic";
```

Comment:
do we need to pass through this additional logic down the callstack? maybe we could simply pass in the `isVerified` key instead with a defaut value being true so that we don't need to update desktop code?

Reply:
this shouldn't affect desktop code since `mode = mnemonic` by default

Reply:
but why this code even need to know about different onboard modes? all it needs is the `isVerified` field